release_notes/ocp-4-8-release-notes.adoc

```diff
@@ -85,6 +85,15 @@ This release adds improvements related to the following components and concepts.
 [id="ocp-4-8-nodes"]
 === Nodes
 
+[id="ocp-4-8-nodes-descheduler-metrics"]
+==== Prometheus metrics for the descheduler
+
+You can now enable Prometheus metrics for the descheduler by adding the `openshift.io/cluster-monitoring=true` label to the `openshift-kube-descheduler-operator` namespace where you installed the descheduler.
+
+The following descheduler metrics are available:
+
+* `descheduler_build_info` - Provides build information about the descheduler.
+* `descheduler_pods_evicted` - Provides details about pods that have been evicted, such as the pod name, namespace, strategy, and result. There must be at least one evicted pod for this metric to appear.
```

Comment:
Also, the description here is not apt as descheduler_pods_evicted does not show details about pods that have been evicted. Below is what it shows:

**descheduler_pods_evicted{endpoint="https", exported_namespace="test1", instance="10.129.3.143:10258", job="metrics", namespace="openshift-kube-descheduler-operator", pod="cluster-6c89d56fdd-5qzrq", result="success", service="metrics", strategy="RemoveDuplicatePods"} | 77**

Reply:
What details do you have in mind?

Reply:
@ingvagabund i think the description should be updated to "provide details about no.of pods that have been evicted in each strategy, namespace, strategy & result. There must be at least one evicted pod for this metric to appear."

Reading the current one user understands that details shown are names of pods that have been evicted where as it is not

Reply:
That's a valid argument. It's true the pod name is not important information in the metrics, nor actually present. The `pod` field corresponds to the descheduler operand pod actually. @bergerhoffer can you drop the "pod name" part?

Reply:
Thanks for the clarification, I think I misunderstood. When I first looked at it, it looked like a single row for each pod that was evicted - I must have missed the number at the end. I'll update.

Reply:
@kasturinarra @ingvagabund I've updated the wording if you can take another look. I went with:

> Provides the number of pods that have been evicted for each strategy, namespace, and result combination.

I added "combination" to be clear that each line was the count for the unique combination of those 3, not a list of all evicted pods in a particular namespace, all evicted pods that failed, etc. Let me know what you think or if there's a better way to get this across. Thanks!

Reply:
I have just opened https://github.com/openshift/cluster-kube-descheduler-operator/pull/186. It's adding another label into the metrics entry, namely `pod_namespace` which indicates the namespace from which the eviction was performed. The original `namespace` indicates the namespace where the descheduler operand is running.

Reply:
Let's avoid mentioning exact labels of the `descheduler_pods_evicted` metric. If it still gets confusing for users, we can update the docs later.

@bergerhoffer no objection here.

Reply:
@ingvagabund Are you saying that you'd rather simplify to something like:

> Provides the number of pods that have been evicted.

Reply:
What's currently mentioned is acceptable. Unless @kasturinarra has other objections.

Reply:
@bergerhoffer one minor comment, do we need the **combination** word in the sentence ? rest looks good to me .

Reply:
Thanks @kasturinarra. I think that "combination" is an important part because it's not showing a count for every strategy and every namespace and every result. It shows a count for each unique combination of the three. I got another writer's opinion and they agreed that "combination" was important, but suggested to move it up earlier in the sentence. So it is now:

> Provides the number of pods that have been evicted for each combination of strategy, namespace, and result.

Let me know if this looks okay to you. Thanks!

Reply:
@ingvagabund should it be "Provides the number of pods that have been evicted for each combination of strategy, pod_namespace, and result" ?

@bergerhoffer looks good to me, once jan confirms it should be namespace itself and not pod_namespace, you could go a head and merge, thanks !!

Reply:
@kasturinarra as discussed earlier in this thread, strategy, namespace, resp. result are not meant to refer to exact names of metrics labels, rather their meaning. The labels can change over time.

> should it be "Provides the number of pods

We can be very specific or we can be more general. Number of pods is a piece of detail. I prefer to be more general so we don't have to update the docs every time we update the metrics.

LGTM

Reply:
Thank you both - merging.